WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -261,87 +261,51 @@
         <%= f.input :base_entry_fee_lowest_denomination, as: :money_amount, currency: @competition.currency_code, disabled: disable_money_input, currency_selector: "#competition_currency_code" %>
 
         <script>
-          var baseFeeUSD = -1;
-
-          // Note: unfortunately we cannot trigger a change on the currency
-          // code update, because we don't have access to the currency
-          // "subunit_to_unit" from the js world.
           function UpdateDues(){
-            <% countryBand = CountryBand.where(iso2: @competition.country_iso2).map(&:number)[0] %>
-            <% if countryBand.present? && countryBand > 0 %>
-              var bandFeeUSD = <%= CountryBand::BANDS[countryBand][:value] %>;
-            <% else %>
-              var bandFeeUSD = 0;
-            <% end %>
-            var pricePerCompetitor = Math.max(bandFeeUSD, baseFeeUSD) * 100;
-            var compLimit = $('select[name="competition[competitor_limit_enabled]"]').val() === "true"
-              ? $('input[name="competition[competitor_limit]"]').val() : null;
-            window.wca.cancelPendingAjaxAndAjax('currency_convert', {
-              url: '<%= currency_convert_path %>',
+
+            var compLimitEnabled = $('select[name="competition[competitor_limit_enabled]"]').val();
+            var compLimit = compLimitEnabled === "true" ? $('input[name="competition[competitor_limit]"]').val() : null;
+            var countryId = $('select[name="competition[countryId]"]').val()
+            var currencyCode = $('select[name="competition[currency_code]"]').val()
+            var entryFee = $('input[name="competition[base_entry_fee_lowest_denomination]"]').val();
+
+            window.wca.cancelPendingAjaxAndAjax('calculate_dues', {
+              url: '<%= calculate_dues_path %>',
               data: {
-                value: compLimit === null ? pricePerCompetitor : pricePerCompetitor * compLimit,
-                from_currency: "USD",
-                to_currency: "<%= @competition.currency_code %>",
+                competitor_limit_enabled: compLimitEnabled,
+                competitor_limit: compLimit,
+                country_id: countryId,
+                currency_code: currencyCode,
+                entry_fee: entryFee,
               },
               success: function(data) {
```

Comment:
I'm not too wild about the entire approach of dynamically computing frontend strings to inject here.
I know this page will be migrated to React soon, but in the meantime your approach will stay around for a few weeks so I'd like to suggest the following alternative:
1. Create a small HTML snippet that is only responsible for showing the Dues estimate textbox in this folder. Be sure to name it with an underscore in front, otherwise the entire page layout will be injected.
2. Have that template do essentially what you're doing now, but as a "real" Rails template and not some "glue and stick stuff together with JS".
3. Have the currency endpoint return that fragment with `render partial: 'your_new_snippet'`
4. Inject the fully rendered HTML fragment into the DOM as-is.

It will change error handling a bit, in that Rails is responsible for showing error strings if anything goes wrong, and then the actual jQuery `error` function below should resort to making the background of the text turn red or something really rudimentary, because that function only concerns real technical errors (like 500 responses) that should only ever happen in 1 / 1000 cases anyways.

See the current `series_eligible_competitions` and `nearby_competitions` controller methods (and corresponding view fragments) for reference. Your view fragment here would be significantly less complex, though.

Taking this approach will also save us from the "i18n within JS that needs comments to be detected by our test suite checks" pain.

Reply:
Closing this because the current approach is better suited for the upcoming React migration